### PR TITLE
show notification bell icon more obviously with role/group to review

### PR DESCRIPTION
**Pull request by jimmytsang**
Essentially this PR featureFlag wraps the following logic ...

- fetchAllRolesToReview for user (new)

- fetchAllGroupsToReview for user (new)

- fetchPendingMembersToReview (existing)

if there are any results from the above fetches , show notification icon like the following 
image

Important files to review are

- HeaderMenu.*js
- redux/*.js
- */api.js

### CODE REVIEW

**Comment by jimmytsang** [comment]
I'm planning on adding functional tests in future pr after the feature works end to end

File: ui/src/components/header/HeaderMenu.js [hunk 1]
```diff
@@ -87,68 +92,103 @@ const HeaderMenuUserDiv = styled.div`
...
+
+    render() {
+        let props = this.props;
+        let reviewNotificationRedDot = '';
+        let icon = 'notification';
+        let clickIcon = this.onClickFunction.bind(
+            this,
+            PageUtils.workflowAdminPage()
+        );
+        if (props.pending || props.reviewRoles || props.reviewGroups) {
```
Comment (MartinTrojans) [inline ui/src/components/header/HeaderMenu.js:137]: Will that be better to put all these conditions together? And if necessary, we can have a function to check the status of pending members, or review roles or groups, and you can just pass these values to that function

**Review by MartinTrojans — APPROVED** [review]

File: ui/src/components/header/HeaderMenu.js (hunk 1 above)
Comment (jimmytsang) [reply ui/src/components/header/HeaderMenu.js:137]: ok good suggestion, i can definitely write a helper function to simplify this to 
```
if (shouldAlertUser(...))
```

File: ui/src/components/header/HeaderMenu.js [hunk 2]
```diff
@@ -87,68 +92,103 @@ const HeaderMenuUserDiv = styled.div`
...
+        this.props.router.push(route, route);
+    }
+
+    componentDidMount() {
+        this.props.getUserPendingMembers();
+        this.props.getReviewRoles();
+        this.props.getReviewGroups();
+        this.api.getPageFeatureFlag('roleGroupReview').then((data) => {
+            this.setState({
+                roleGroupReviewFeatureFlag: data['roleGroupReviewFeatureFlag'],
```
Comment (chandrasekhar1996) [inline ui/src/components/header/HeaderMenu.js:124]: should we add this roleGroupReviewFeatureFlag to default-config.js

File: ui/src/components/header/HeaderMenu.js [hunk 3]
```diff
@@ -87,68 +92,103 @@ const HeaderMenuUserDiv = styled.div`
...
-                </Menu>
-            </HeaderMenuUserDiv>
-            {props.headerDetails && props.headerDetails.userId && (
+    }
+
+    render() {
+        let props = this.props;
+        let reviewNotificationRedDot = '';
+        let icon = 'notification';
+        let clickIcon = this.onClickFunction.bind(
```
Comment (chandrasekhar1996) [inline ui/src/components/header/HeaderMenu.js:133]: future note: when you add the new tabs for role/group review, update this to go to the tab which has pending approval work.

**Review by chandrasekhar1996 — CHANGES_REQUESTED** [review]

File: ui/src/components/header/HeaderMenu.js (hunk 3 above)
Comment (jimmytsang) [reply ui/src/components/header/HeaderMenu.js:133]: oh interesting good suggestion

File: ui/src/components/header/HeaderMenu.js (hunk 1 above)
Comment (jimmytsang) [reply ui/src/components/header/HeaderMenu.js]: done.

File: ui/src/components/header/HeaderMenu.js (hunk 2 above)
Comment (jimmytsang) [reply ui/src/components/header/HeaderMenu.js:130]: done

**Review by chandrasekhar1996 — APPROVED** [review]

**Pull request merged**